### DISCUSSION TITLE
Is there a way to trigger data fetcher if a specific field is requested irrespective of it's parent

### DISCUSSION THREAD

**Discussion by arunnaparna** [Netflix/dgs-framework#532] [Q&A] [answered]
I would like to trigger a data fetcher when the request is for a specific attribute/field. This field can be part of multiple Types. 

e.g., 

```
interface SourceInformation{
  fromA : String
  fromB: String
}

type One implements SourceInformation {
   name: String
   fromA : String
   fromB: String
}
type Two  implements SourceInformation{
   name: String
   fromA : String
   fromB: String
}

type Request {
    one: One
    two: Two
}
```

With this structure, if the request is for `fromA`, can we trigger a data fetcher? This field can be part of either `One` or `Two` types. 
I have gone through the section Nested Data fetchers in https://netflix.github.io/dgs/advanced/context-passing/ 
Can the framework handle such usecase? or should it be handled explicitly using: 
`datafetchingenvironment.getSelectionSet().contains("fromA")`

**Comment by paulbakker** [comment, marked as answer]
I might be misunderstanding the question. But here's an attempt at an answer. You can just define the datafetcher for `fromA` on either the interface type or the concrete types.

```
@DgsData(parentType="SourceInformation")
public String fromA() {

}
```

This would define a single datafetcher that works for both `One` and `Two`. Alternatively you can create the datafetcher for the implementing type explicitly:

```
@DgsData(parentType="One")
public String fromA() {

}

@DgsData(parentType="Two")
public String fromA() {

}
```
